test/tests/functional/pbs_multi_sched.py

```diff
@@ -0,0 +1,573 @@
+
+from tests.functional import *
+import shutil
+
+
+@tags('multisched')
+class TestMultipleSchedulers(TestFunctional):
+
+    """
+    Test suite to test different scheduler interfaces
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        self.server.manager(MGR_CMD_CREATE, SCHED,
+                            {'partition': 'P1,P4'},
+                            id="sc1")
+        self.sched_configure('sc1')
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 1}, id="sc1")
+
+        self.server.manager(MGR_CMD_CREATE, SCHED,
+                            {'partition': 'P2',
+                             'sched_priv':
+                             '/var/spool/pbs/sched_dir/sched_priv_sc2',
+                             'sched_log':
+                             '/var/spool/pbs/sched_dir/sched_logs_sc2'},
+                            id="sc2")
+        self.sched_configure('sc2', '/var/spool/pbs/sched_dir')
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 1}, id="sc2")
+
+        self.server.manager(MGR_CMD_CREATE, SCHED,
+                            {'partition': 'P3'},
+                            id="sc3")
+        self.sched_configure('sc3')
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 1}, id="sc3")
+
+        a = {'queue_type': 'execution',
+             'started': 't',
+             'enabled': 't'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq2')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq3')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq4')
+        p1 = {'partition': 'P1'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p1, id='wq1')
+        p2 = {'partition': 'P2'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p2, id='wq2')
+        p3 = {'partition': 'P3'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p3, id='wq3')
+        p4 = {'partition': 'P4'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p4, id='wq4')
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes('vnode', a, 5, self.mom)
+        self.server.manager(MGR_CMD_SET, NODE, p1, id='vnode[0]', expect=True)
+        self.server.manager(MGR_CMD_SET, NODE, p2, id='vnode[1]', expect=True)
+        self.server.manager(MGR_CMD_SET, NODE, p3, id='vnode[2]', expect=True)
+        self.server.manager(MGR_CMD_SET, NODE, p4, id='vnode[3]', expect=True)
+
+    def sched_configure(self, sched_name, sched_home=None):
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        if sched_home is None:
+            sched_home = pbs_home
+        sched_priv_dir = 'sched_priv_' + sched_name
+        sched_logs_dir = 'sched_logs_' + sched_name
+        if not os.path.exists(os.path.join(sched_home, sched_priv_dir)):
+            shutil.copytree(os.path.join(pbs_home, 'sched_priv'),
```

Comment:
@visheshh 
You are using "shutil" at many places.
Recommend to use dshutil or enhance it as per your use. As dshutil provides ssh, scp facility. It will more useful.
Suppose someone wants to test multisched running on remote host other than the serverhost, dshutil is useful.

Reply:
@varunsonkar , I will use dshutil , Only one issue is I have to do a mkdir if the path does not exist and do run_copy.
But anyhow the interface will change with next implementation

Reply:
@visheshh 
As far as I know even with next implementation the Admin still will have to create the home and exec directories for multi-sched.
So I believe my comment holds even for next implementation.
@suresh-thelkar ,
Please let me know if this is not the case with the next implementation.